src/server/redis_connection.h

```diff
@@ -93,6 +93,7 @@ class Connection : public EvbufCallbackBase<Connection> {
   int GetListeningPort() const { return listening_port_; }
   void SetAnnounceIP(std::string ip) { announce_ip_ = std::move(ip); }
   std::string GetAnnounceIP() const { return !announce_ip_.empty() ? announce_ip_ : ip_; }
+  uint32_t GetAnnouncePort() const { return listening_port_ != 0 ? listening_port_ : port_; }
   std::string GetAnnounceAddr() const { return GetAnnounceIP() + ":" + std::to_string(listening_port_); }
```

Comment:
We should fix the `GetAnnounceAddr` as well.

Reply:
I had the same idea two days ago, but I am not very familiar with Kvrcoks source code. I have been new to Kvrocks for almost 2 months. I'm not sure if I fix GetAnnounceAddr, other calls will have a side effect or not. So I only added GetAnnouncePort for securely in case of bad import.

Reply:
@xiaofan8421 That's fine, can fix GetAnnounceAddr's listening part as well.

Reply:
ok, done. Please review it.
Moreover, maybe GetAnnouncePort interface should be reserved for replica-announce-port or something else in the future.